app/models/manageiq/providers/vmware/infra_manager/inventory/collector.rb

```diff
@@ -19,25 +19,64 @@ def run
 
     saver.start_thread
 
-    until exit_requested
-      monitor_updates
-    end
+    vim = connect
+    property_filter = create_property_filter(vim)
 
-    saver.stop_thread
+    version = initial_refresh(vim, property_filter)
+    _log.info("Refreshing initial inventory...Complete")
+
+    until exit_requested
+      persister = targeted_persister_klass.new(ems)
+      version = monitor_updates(vim, property_filter, version, persister)
+    end unless run_once
 
     _log.info("Monitor updates thread exited")
+  ensure
+    saver.stop_thread
+    destroy_property_filter(property_filter)
+    disconnect(vim)
   end
 
   def stop
     _log.info("Monitor updates thread exiting...")
     self.exit_requested = true
   end
 
-  def monitor_updates
-    vim = connect
-    wait_for_updates(vim)
-  ensure
-    disconnect(vim)
+  def initial_refresh(vim, property_filter)
+    # Send the "special initial data version" i.e. an empty string
+    # so that we get all inventory back in the first update set
+    version = ""
+
+    # Use the full refresh persister for the initial UpdateSet from WaitForUpdates
```

Comment:
either scrap comment or put in method comment.

not needed since you separated the 2 calls to `monitor_update` and don't re-initialize this variable